ledger/eval.go

```diff
@@ -801,6 +804,17 @@ func (eval *BlockEvaluator) transaction(txn transactions.SignedTxn, evalParams *
 		}
 	}
 
+	// in case of a CompactCertTx transaction, we want to "apply" it only in validate or generate mode. This will deviate the cow's CompactCertNext depending of
+	// whether we're in validate/generate mode or not, however - given that this variable in only being used in these modes, it would be safe.
+	// The reason for making this into an exception is that during initialization time, the accounts update is "converting" the recent 320 blocks into deltas to
+	// be stored in memory. These deltas don't care about the compact certificate, and so we can improve the node load time. Additionally, it save us from
+	// performing the validation during catchup, which is another performance boost.
+	if (eval.validate || eval.generate) && txn.Txn.Type == protocol.CompactCertTx {
```

Comment:
It seems like a natural place for this code would be in the `case protocol.CompactCertTx` branch inside of `applyTransaction()`; why not put it there?  Is it because `applyTransaction()` is not a method on `BlockEvaluator`?  It doesn't seem to be used by anything except `BlockEvaluator.transaction()`, so should be reasonable to make it a method so that we can access `eval.validate` and `eval.generate` there..

Reply:
that's a good point. I'll make that change.